app/main/browser-window.js

```diff
@@ -71,15 +85,39 @@ export async function createBrowserWindow(userAgentClient, onload) {
     browser.webContents.send('user-agent-service-info', { port, host, version });
   });
 
+  // Only check and send the set default browser message if this is the first
+  // window opened, we're in a packaged build, on OSX (for now), and this isn't
+  // already the default browser.
+  const shouldCheckDefaultBrowser = browserWindows.length === 0 &&
+                                    process.env.NODE_ENV === 'production' &&
+                                    process.platform === 'darwin' &&
+                                    !protocols.isDefaultBrowser();
+
   // 'window-ready' is called if an error occurred loading the client, or once
   // the client has connected to the User Agent Service correctly.
   browser.once('window-ready', error => {
+    browser.IS_READY = true;
```

Comment:
We have a Deferred class in ui/shared/util/deferred.js, seems like it would work better than IS_READY booleans. Should move that in the app/shared and use it here.